pydantic/_internal/_core_utils.py

```diff
@@ -65,7 +65,7 @@ def get_type_ref(type_: type[Any], args_override: tuple[type[Any], ...] | None =
     when creating generic models without needing to create a concrete class.
     """
     origin = type_
-    args = args_override or ()
+    args = type_.__args__ if isinstance(type_, _GenericAlias) else (args_override or ())  # type: ignore
```

Comment:
@dmontagu, could you please explain why we changed this line?

I understand the value of the `deepcopy` on the model's fields in `_generate_schema.py`, but I don't quite understand the purpose of this line here.

Reply:
I just noticed this misbehavior — right now, it's not producing a type ref that reflects the generic parameters like it is supposed to. In particular, this means that in principle you could get two different schemas with the same ref but that should have different parameters. I don't know why that doesn't seem to be affecting the validation behavior, but I do think this is an improvement either way.